### PR TITLE
Add docs and demo for imageTags

### PR DESCRIPTION
**Pull request by Liujingfang1**
#193

### CODE REVIEW

File: docs/kustomization.yaml [hunk 1]
```diff
@@ -184,3 +184,20 @@ vars:
       apiVersion: apps/v1
     fieldref:
       fieldpath: spec.template.spec.restartPolicy
+
+# ImageTags are used to modify the tags for images used in different resources without creating patches.
```
Comment (monopole) [inline docs/kustomization.yaml]: ImageTags modify the tags for images without creating patches.
E.g. Given this fragment of a Deployment:

File: docs/kustomization.yaml [hunk 2]
```diff
@@ -184,3 +184,20 @@ vars:
...
+# For example, in deployment.yaml you have following container
+# ```
+#  containers:
+#    - name: myapp
+#      image: mycontainerregistry/myimage:v0
+#    - name: nginxapp
+#      image: nginx:1.7.9
+#```
+# You want to change the tag of myimage to v1 and change the tag of nginx to 1.8.0. You can add following
+# imageTags in kusotmization.yaml. Then kustomize build will replace the old tags with the new tags.
```
Comment (monopole) [inline docs/kustomization.yaml]: one can change the tag of myimage to v1 and the tag of nginx to 1.8.0 with
the following:

File: examples/imageTags.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+Add `imageTags`:
+<!-- @addImageTags @test -->
+```
+cd $DEMO_HOME
+kustomize edit set imageTag busybox:1.29.1
+```
+
+The `kustomization.yaml` will be added following `imageTags`.
```
Comment (monopole) [inline examples/imageTags.md]: Indent code that isn't meant as a command, e.g.
> ```
> imageTags:
> - name: busybox
> - newTag: 1.29.1
> ```

File: examples/imageTags.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+```
+imageTags:
+- name: busybox
+  newTag: 1.29.1
+```
+
+Now build this `kustomization`
+<!-- @kustomizeBuild @test -->
+```
+kustomize build $DEMO_HOME
```
Comment (monopole) [inline examples/imageTags.md]: Confirm that this replaces _both_ busybox tags:

```
test 2 == \
  $(kustomize build $DEMO_HOME | grep busybox:1.29.1 | wc -l); \
  echo $?
```

File: examples/imageTags.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+The image tag in the output is updated to `1.29.1`, both in the initContainers and containers.
+
+```
```
Comment (monopole) [inline examples/imageTags.md]: I don't think that showing the output is necessary, but if you want to, indent it to distinguish it from long command blocks.

File: examples/imageTags.md (hunk 4 above)
Comment (Liujingfang1) [reply examples/imageTags.md]: This is added

File: docs/kustomization.yaml (hunk 2 above)
Comment (Liujingfang1) [reply docs/kustomization.yaml]: done

**Pull request merged**